explainer.md

```diff
@@ -0,0 +1,802 @@
+
+ZonedDateTime represents an OffsetDateTime combined with an *IANA Timezone*. With that added information ZonedDateTime is able to observe daylight savings rules.
+
+### Instant
+An `Instant` is an object that specifies a specific point in time. For convenience of interoperability it uses *nanoseconds since the unix-epoch* to do so.
+#### Instant() - constuctor
+
+#### Instant.prototype.epochSeconds : number
+
+The `epochSeconds` property of an `Instant` object is readonly and represents the *seconds
+since unix-epoch*.
+
+#### Instant.prototype.epochMilliseconds: number
+
+The `epochMilliseconds` property of an `Instant` object is readonly and represents the *milliseconds
+since unix-epoch*.
+
+#### Instant.prototype.epochMicroseconds: BigInt
+
+The `epochMicroseconds` property of an `Instant` object is readonly and represents the *microseconds
+since unix-epoch*.
+
+#### Instant.prototype.epochNanoseconds: BigInt
+
+The `epochNanoseconds` property of an `Instant` object is readonly and represents the *nanoseconds
+since unix-epoch*.
+
+#### Instant.prototype.withZone(zone: string) : ZonedDateTime
+
+This creates a `ZonedDateTime` by applying a *iana timezone* to the instant.
+
+This is equivalent to `new ZonedDateTime(instant, zone)`
+
+#### Instant.prototype.withOffset(offset: string) : OffsetDateTime
+
+This creates a `OffsetDateTime` by applying an *offset-string* to the instant.
+
+This is equivalent to `new OffsetDateTime(instant, offset)`
+
+#### Instant.fromString(iso: string) : Instant
+
+Parses a `string` that must be in the same ISO-8601 format as produced by `Instant.prototype.toString()` and creates a new `Instant` object from it.
+
+#### Instant.fromEpochSeconds(epochseconds: number) : Instant
+
+Equivalent to `Instant.fromMilliseconds(seconds * 1000)`.
+
+#### Instant.fromEpochMilliseconds(epochmillis: number) : Instant
+
+Equivalent to `Instant.fromEpochMicroseconds(BigInt(epochmillis) * 1000n)`.
+
+#### Instant.fromEpochMicroseconds(epochmicros: BigInt) : Instant
+
+Equivalent to `Instant.fromEpochNanoseconds(epochmicros * 1000n)`.
+
+#### Instant.fromEpochNanoseconds(epochnanos: BigInt) : Instant
+
+Equivalent to `new Instant(epochnanos)`.
+
+### OffsetDateTime
+
+An `OffsetDateTime` is an object that specifies a specific point in time with a specific offset from *UTC*.
+It bases this on an `Instant` and an offsett `string`.
+
+#### OffsetDateTime(instant: Instant, offset: string) - constructor
+
+The constructor may only be called as such. It takes two arguments. The first is an instance
+of `Instant` and the second is a `string` representing a valid time offset.
+
+#### OffsetDateTime.prototype.instant : BigInt
+
+The instance of `Instant` that was used to create this `OffsetDateTime`.
+
+#### OffsetDateTime.prototype.offset : string
+
+The time-zone offset represented as a `string`.
```

Comment:
I was hoping to. They were created/useful only in misshapen timezone handling in `Date`. I consider them a bad relic. (Convince me otherwise 😄 )

Reply:
I don't know if I can convince you, but I imagine that _not_ having it will create cases where authors manually perform awkward calculations like `offsetMinutes = dateTime.offset.split(":").reduce((sum, part, i) => (sum || -0) + Math.sign(Infinity/sum) * part * 60**(1-i), 0)` or `offsetSeconds = CivilDateTime.from(dateTime).withTimeZone("Etc/UTC").instant.epochSeconds - dateTime.instant.epochSeconds`, although we might be able to mitigate that with `minus`.